actors/user/src/main/java/org/sunbird/user/actors/UserRoleActor.java

```diff
@@ -0,0 +1,355 @@
+    Response rolegroup =
+        cassandraOperation.getAllRecords(
+            roleGroupDbInfo.getKeySpace(), roleGroupDbInfo.getTableName());
+    Response urlAction =
+        cassandraOperation.getAllRecords(
+            urlActionDbInfo.getKeySpace(), urlActionDbInfo.getTableName());
+    List<Map<String, Object>> urlActionListMap =
+        (List<Map<String, Object>>) urlAction.getResult().get(JsonKey.RESPONSE);
+    List<Map<String, Object>> roleGroupMap =
+        (List<Map<String, Object>>) rolegroup.getResult().get(JsonKey.RESPONSE);
+    list = (List<Map<String, Object>>) response.getResult().get(JsonKey.RESPONSE);
+    if (list != null && !(list.isEmpty())) {
+      // This map will have all the master roles
+      for (Map<String, Object> map : list) {
+        Map<String, Object> roleResponseMap = new HashMap<>();
+        roleResponseMap.put(JsonKey.ID, map.get(JsonKey.ID));
+        roleResponseMap.put(JsonKey.NAME, map.get(JsonKey.NAME));
+        List<String> roleGroup = (List) map.get(JsonKey.ROLE_GROUP_ID);
+        List<Map<String, Object>> actionGroupListMap = new ArrayList<>();
+        roleResponseMap.put(JsonKey.ACTION_GROUPS, actionGroupListMap);
+        Map<String, Object> subRoleResponseMap = null;
+        for (String val : roleGroup) {
+          subRoleResponseMap = new HashMap<>();
+          Map<String, Object> subRoleMap = getSubRoleListMap(roleGroupMap, val);
+          List<String> subRole = (List) subRoleMap.get(JsonKey.URL_ACTION_ID);
+          List<Map<String, Object>> roleUrlResponList = new ArrayList<>();
+          subRoleResponseMap.put(JsonKey.ID, subRoleMap.get(JsonKey.ID));
+          subRoleResponseMap.put(JsonKey.NAME, subRoleMap.get(JsonKey.NAME));
+          for (String rolemap : subRole) {
+            roleUrlResponList.add(getRoleAction(urlActionListMap, rolemap));
+          }
+          if (subRoleResponseMap.containsKey(JsonKey.ACTIONS)) {
+            List<Map<String, Object>> listOfMap =
+                (List<Map<String, Object>>) subRoleResponseMap.get(JsonKey.ACTIONS);
+            listOfMap.addAll(roleUrlResponList);
+          } else {
+            subRoleResponseMap.put(JsonKey.ACTIONS, roleUrlResponList);
+          }
+          actionGroupListMap.add(subRoleResponseMap);
+        }
+
+        resposnemap.add(roleResponseMap);
+      }
+    }
+    mergeResponse.getResult().put(JsonKey.ROLES, resposnemap);
+    sender().tell(mergeResponse, self());
+  }
+
+  /**
+   * This method will assign roles to users or user organizations.
+   *
+   * @param actorMessage
+   */
+  @SuppressWarnings("unchecked")
+  private void assignRoles(Request actorMessage) {
+    Map<String, Object> requestMap = actorMessage.getRequest();
+
+    if (null != requestMap.get(JsonKey.ROLES)
+        && !((List<String>) requestMap.get(JsonKey.ROLES)).isEmpty()) {
+      String msg = Util.validateRoles((List<String>) requestMap.get(JsonKey.ROLES));
```

Comment:
Create a RoleService with method validateRoles which throws ProjectCommonException if there is any invalid role.